### PR TITLE
server: skip the engine key when match store label

### LINKED ISSUE #8480: replication.strictly-match-label=true &  engine=tiflash have Incompatibility problem
### What did you do?

1. After expanding the tiflash, the tiflash node will automatically have a layer of label information: engine: "tiflash"
2, When the pd is configured: replication. strictly-match-label: true
3, This causes tiflash unable to start because replication. location-labels in the default pd config does not exist: engine layer configuration.

### What did you expect to see?
1、tiflash start successfully
2、replication. location-labels can automatically recognize and add  a new "engine" layer;

### What did you see instead?
1、tiflash start failed；
2、replication. location-labels can not automatically recognize and  keep old layer;

### What version of PD are you using (`pd-server -V`)?

Release Version: v7.5.0
Edition: Community
Git Commit Hash: ef6ba8551e525a700546d6bdb7ad6766115209cc
Git Branch: heads/refs/tags/v7.5.0
UTC Build Time:  2023-11-16 10:31:04

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #8480.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

**Review by okJiang — APPROVED** [review]
LGTM, But `LocationLabels` seems to have been designed to be equivalent to `Labels` (if not, ignore it), but now there are other concepts for `Labels`, such as `engine`. So it's not clear if there are other places where this gap exists

File: server/api/label_test.go [hunk 1]
```diff
@@ -277,19 +278,47 @@ func (suite *strictlyLabelsStoreTestSuite) TestStoreMatch() {
...
+					{
+						Key:   core.EngineKey,
+						Value: core.EngineTiFlash,
+					},
+				},
+				Version: "3.0.0",
+			},
+			valid:       true,
+			expectError: "placement rules is disabled",
+		},
```
[the remark this answers is not in the document]
Comment (rleungx) [reply server/api/label_test.go:304]: I think yes, cuz we cannot add a TiFlash store when the placement rule is set to disable.

[the remark this answers is not in the document]
Comment (rleungx) [reply server/api/label_test.go:304]: There is no guarantee that LocationLabels should be equivalent to Labels.

**Review by nolouch — APPROVED** [review]

**Pull request merged**